man/org.freedesktop.login1.xml

```diff
@@ -992,6 +992,14 @@ node /org/freedesktop/login1/user/_1000 {
     methods:
       Terminate();
       Kill(in  i signal_number);
+      SecureLock();
```

Comment:
adding a concept such as a "SecureLock" might be OK, but it kinda irks me that our screen lock concept is currently so weird, i.e. we only fire a signal, that's it. ideally we'd track the screen lock status, and allow clients to wait for the screen lock to be activated.

it seems to me that for the secure lock we'd provide features we should also provide for the screen lock.

generally i think distinguishing between two levels of lock, i.e. screen lock and secure lock makes sense. I just wish they would expose more similar semantics.

Reply:
I agree that normal screen locks should be improved. Unfourtunately, I'm not sure if there's a great way to do that in a backwards-compatible way. And anyway, I'm just not sure that is necessarily actionable for this PR. Screen locks are per-session concepts, but secure locks are per-user concepts

What do you want me to implement re: screen locks in this PR?

Reply:
it's fine to not hack anything up for screen locks right now, all i am asking for to keep the concept in mind.

my original request was kinda in the hope we could have a common set of functions that just takes an enum or flags, asking for one of the two kinds.

but i figure this doesn't actually make much sense, since the screen locks lock session objects, while the "secure locks" lock users as a whole.

hence maybe ignore me.